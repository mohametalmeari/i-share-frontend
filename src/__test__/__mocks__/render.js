import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../../redux/store';

const customRender = (ui, options) => render(ui, {
  wrapper: ({ children }) => <Provider store={store}>{children}</Provider>,
  ...options,
});

export default customRender;