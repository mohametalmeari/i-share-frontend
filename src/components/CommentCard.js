import PropTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
import { useEffect, useState } from 'react';
import {
  deleteComment, fetchComments, fetchReplies, likeComment,
} from '../redux/posts/commentSlice';
import ReplyCard from './ReplyCard';
import ReplyForm from './ReplyForm';

const CommentCard = ({
  id, photoId, name, content, likes, liked, control,
}) => {
  const [showReply, setShowReply] = useState({ form: false, replies: false });
  const dispatch = useDispatch();
  const handleDelete = async () => {
    await dispatch(deleteComment({ id, photoId }));
    dispatch(fetchComments(photoId));
  };

  const { replies } = useSelector((state) => state.comment);
  useEffect(() => {
    dispatch(fetchReplies({ photoId, commentId: id }));
    console.log('replies', replies);
    console.log('-----');
  }, []);

  const toggleShowReplyForm = () => {
    setShowReply({ ...showReply, form: !showReply.form });
  };

  const toggleShowReplies = () => {
    setShowReply({ ...showReply, replies: !showReply.replies });
  };
  const handleLike = async () => {
    dispatch(likeComment({ id, photoId, liked }));
  };
  return (
    <>
      {id}
      ,
      {name}
      :
      {content}
      ,
      {likes}
      ,
      <button type="button" onClick={handleLike}>
        {liked ? 'Unlike' : 'Like'}
      </button>

      <button type="button" onClick={toggleShowReplyForm}>
        Reply from
      </button>
      {showReply.form && (
        <ReplyForm
          photoId={photoId}
          commentId={id}
          setShowReply={setShowReply}
        />
      )}

      <button type="button" onClick={toggleShowReplies}>
        Show replies
      </button>
      {showReply.replies && replies.map((reply) => (
        <ReplyCard
          key={reply.id}
          id={reply.id}
          photoId={photoId}
          commentId={id}
          name={reply.user.name}
          content={reply.content}
          likes={reply.likes}
          liked={reply.liked}
          control={reply.user.control}
        />
      ))}

      {control && (
      <button type="button" onClick={handleDelete}>
        Delete comment
      </button>
      )}
      <br />
    </>
  );
};

export default CommentCard;

CommentCard.propTypes = {
  id: PropTypes.number.isRequired,
  photoId: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  content: PropTypes.string.isRequired,
  likes: PropTypes.number.isRequired,
  liked: PropTypes.bool.isRequired,
  control: PropTypes.bool.isRequired,
};
