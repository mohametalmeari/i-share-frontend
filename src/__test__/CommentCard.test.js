import render from 'render';
import CommentCard from '../components/CommentCard';

describe(CommentCard, () => {
  const comment = {
    id: 123,
    user: {name: 'Alex', control: false},
    content: 'Test Comment',
    likes: 7,
    replies: 13,
    liked: true,
  }
  const photoId = 456;

  test('renders Snapshot', () => {
    const { container } = render(
        <CommentCard
          id={comment.id}
          photoId={photoId}
          name={comment.user.name}
          content={comment.content}
          likes={comment.likes}
          repliesCount={comment.replies}
          liked={comment.liked}
          control={comment.user.control}
        />
    );
    expect(container).toMatchSnapshot();
  });
});