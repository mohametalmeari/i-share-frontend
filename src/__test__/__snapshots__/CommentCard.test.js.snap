// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentCard renders Snapshot 1`] = `
<div>
  <div
    class="comment-container"
  >
    <div>
      <b>
        Alex
        : 
      </b>
      <span>
        Test Comment
      </span>
    </div>
    <div
      class="interaction-counts"
    >
      <span>
        7 Likes, 
      </span>
      <span>
        13 Replies
      </span>
    </div>
    <div
      class="comment-interact-container"
    >
      <button
        class="link comment-interact"
        type="button"
      >
        Unlike
      </button>
       - 
      <button
        class="link comment-interact"
        type="button"
      >
        Reply
      </button>
       - 
      <button
        class="link comment-interact"
        type="button"
      >
        Show replies
      </button>
    </div>
    <div
      class="replies-container"
    />
  </div>
</div>
`;
